### PR TITLE
[bug] get armada object after valid document

### LINKED ISSUE #128: Bug: Document validation is performed after attempting to use the documents
**Is this a bug report or feature request?** (choose one):

**Python Version** (output of `python --version`):

**Development or Deployment Environment?**:

**Release Tag or Master**:

**Expected Behavior**: we should run the `self.get_armada_manifest()` method in pre-flight after the document validation has occurred, rather than in the constructor

**What Actually Happened**: the manifest is created using `self.documents` in the constructor which bypasses the linting/validation process. therefore, if there is a problem, it will cause unexpected behavior rather than being caught in the linting process.

**How to Reproduce the Issue** (as minimally as possible):

**Any Additional Comments**:
@gardlt

### PR DESCRIPTION
**Pull request by gardlt**
- calling object get after vaild doc

**What is the purpose of this pull request?**:

**What issue does this pull request address?**: Fixes #128 

**Notes for reviewers to consider**:

**Specific reviewers for pull request**:

### CODE REVIEW

File: armada/handlers/armada.py [hunk 1]
```diff
@@ -66,8 +66,8 @@ def __init__(self,
         self.wait = wait
         self.timeout = timeout
         self.tiller = Tiller(tiller_host=tiller_host, tiller_port=tiller_port)
-        self.documents = yaml.safe_load_all(file)
-        self.config = self.get_armada_manifest()
+        self.documents = list(yaml.safe_load_all(file))
```
Comment (theyer) [inline armada/handlers/armada.py]: @gardlt why the list wrapper?

Comment (gardlt) [inline armada/handlers/armada.py]: when we use a yaml.load it creates an object that can be lazy iterated on and its kinda a one off, we need to cast it to a list so we can make it iterate. .

Comment (theyer) [inline armada/handlers/armada.py]: @gardlt got it. current problem is the rest of the code isn't expecting it to be in a list. can we say something like `self.documents = list(yaml.safe_load_all(file))[0]` to get the values from the generator but store it as a dict as expected?

Comment (theyer) [inline armada/handlers/armada.py]: ah nevermind i misunderstood how the list was working but `apply` doesn't work with this change

Comment (theyer) [inline armada/handlers/armada.py]: alright sorry that was on my end. lgtm

**Pull request merged**